std=c99 required by gcc 4.8.5

I suspect that you may already be aware of the following, but if not perhaps it is useful to know:

I get the following errors and suggestions from gcc (4.8.5) when compiling hstlib, samtools or bcftools:

```
vcf.c:5109:5: error: 'for' loop initial declarations are only allowed in C99 mode
     for (int idx = 0; idx < sizeof(errdesc_bcf) / sizeof(err_desc); ++idx) {
     ^
vcf.c:5109:5: note: use option -std=c99 or -std=gnu99 to compile your code
```

they are solved by adding -std=gnu99 to the CFLAGS of the Makefiles of hstlib, samtools and bcftools.

I presume that this is specific to the version of gcc and hence not that important. Especially as it seems like there is a plan to specify std=c99 from looking at the htslib Makefile.

So this does not seem to be a problem per se, but I'm wondering whether there there is any potential problem with adding std=c99 or std=gnu99?

cheers,

Martin

--- a/.cirrus.yml
+++ b/.cirrus.yml
@@ -137,6 +137,7 @@ rocky_task:
     LC_ALL: C
     CIRRUS_CLONE_DEPTH: 1
     USE_CONFIG: yes
+    CFLAGS: -std=gnu90
 
   # NB: we could consider building a docker image with these
   # preinstalled and specifying that instead, to speed up testing.

--- a/vcf.c
+++ b/vcf.c
@@ -5097,6 +5097,7 @@ static int add_desc_to_buffer(char *buffer, size_t *offset, size_t maxbuffer, co
 const char *bcf_strerror(int errorcode, char *buffer, size_t maxbuffer) {
     size_t usedup = 0;
     int ret = 0;
+    int idx;
 
     if (!buffer || maxbuffer < 4)
         return NULL;           //invalid / insufficient buffer
@@ -5106,7 +5107,7 @@ const char *bcf_strerror(int errorcode, char *buffer, size_t maxbuffer) {
         return buffer;
     }
 
-    for (int idx = 0; idx < sizeof(errdesc_bcf) / sizeof(err_desc); ++idx) {
+    for (idx = 0; idx < sizeof(errdesc_bcf) / sizeof(err_desc); ++idx) {
         if (errorcode & errdesc_bcf[idx].errorcode) {    //error is set, add description
             ret = add_desc_to_buffer(buffer, &usedup, maxbuffer, errdesc_bcf[idx].description);
             if (ret < 0)

